### PR TITLE
Temporal: Split up ZonedDateTime/old/string-parsing.js and move out of staging

### PR DESCRIPTION
**Pull request by catamorphism**

### CODE REVIEW

File: test/built-ins/Temporal/ZonedDateTime/from/reject-throws-if-offset-does-not-match-iana-time-zone.js
Comment (ptomato) [inline test/built-ins/Temporal/ZonedDateTime/from/reject-throws-if-offset-does-not-match-iana-time-zone.js:1]: This one should probably be combined with `offset-does-not-match-iana-time-zone.js` which you ported in one of your earlier PRs.

**Review by ptomato — CHANGES_REQUESTED** [review]
Naming nitpick: Let's prefix the ones specifically about string parsing with `argument-string-`, to match the existing string parsing tests in this folder. e.g. `decimal-places.js` → `argument-string-decimal-places.js`.

We should also expand these to cover all API entry points that parse a string, but I'll add that to the list of followups.

File: test/built-ins/Temporal/ZonedDateTime/from/reject-throws-if-offset-does-not-match-iana-time-zone.js
Comment (catamorphism) [reply test/built-ins/Temporal/ZonedDateTime/from/reject-throws-if-offset-does-not-match-iana-time-zone.js:1]: Done.

**Review by ptomato — APPROVED** [review]
LGTM, thanks!

**Pull request merged**